include/EASTL/algorithm.h

```diff
@@ -976,23 +980,59 @@ namespace eastl
 	/// Complexity: Either two or three comparisons will be required, depending 
 	/// on the values.
 	///
+	template <typename T>
+	inline T&& median(T&& a, T&& b, T&& c)
+	{
+		return eastl::forward<T>(median_impl<T&&>(eastl::forward<T>(a), eastl::forward<T>(b), eastl::forward<T>(c)));
```

Comment:
you shouldn't need the explict template parameter for median_impl here as it will be deduced from the forwarded function parameters